tests/zfs-tests/tests/functional/write_dirs/setup.ksh

```diff
@@ -51,4 +51,7 @@ DISK=${DISKS%% *}
 
 log_must set_partition $SLICE "" $SIZE $DISK
 
+if is_linux; then
+	export SLICE=1
+fi
```

Comment:
OK, got it.  Then this LGTM.  You got really unlucky and hit a bunch of unrelated issues in the automated but nothing to caused by this.  I'll merge it shortly.